lib/google/cloud/gemserver/cli/server.rb

```diff
@@ -106,11 +106,15 @@ def delete proj_id
           # @private Creates a key with all permissions and sets it in the
           # necessary configurations (gem credentials and bundle config).
           def setup_default_keys
-            puts "Would you like to setup a default key? [Y/n] (default yes)"
-            should_create = user_input
+            should_create = user_input("Would you like to setup a default " \
+              "key? [Y/n] (default yes)")
             return if should_create.downcase == "n"
             gemserver_url = remote
-            key = extract_key(Request.new(gemserver_url).create_key)
+            res = Request.new(gemserver_url).create_key
+            abort "Error generating key" if res.include? "Internal Server Error"
```

Comment:
Is it possible to check the HTTP response status instead of scan for a particular string?